src/sentry/api/endpoints/project_symbol_sources.py

```diff
@@ -0,0 +1,353 @@
+        redacted = redact_source_secrets(sources)
+
+        if id:
+            for source in redacted:
+                if source["id"] == id:
+                    return Response([source])
+            return Response(data={"error": f"Unknown source id: {id}"}, status=404)
+
+        return Response(redacted)
+
+    @extend_schema(
+        operation_id="Delete a Symbol Source from a Project",
+        parameters=[
+            GlobalParams.ORG_SLUG,
+            GlobalParams.PROJECT_SLUG,
+            ProjectParams.source_id("The ID of the source to delete.", True),
+        ],
+        responses={
+            204: RESPONSE_NO_CONTENT,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+        examples=ProjectExamples.DELETE_SYMBOL_SOURCE,
+    )
+    def delete(self, request: Request, project: Project) -> Response:
+        """
+        Delete a custom symbol source from a project.
+        """
+        id = request.GET.get("id")
+        custom_symbol_sources_json = project.get_option("sentry:symbol_sources") or []
+
+        sources = parse_sources(custom_symbol_sources_json, False)
+
+        if id:
+            filtered_sources = [src for src in sources if src["id"] != id]
+            if len(filtered_sources) == len(sources):
+                return Response(data={"error": f"Unknown source id: {id}"}, status=404)
+
+            serialized = json.dumps(filtered_sources)
+            project.update_option("sentry:symbol_sources", serialized)
+            return Response(status=204)
+
+        return Response(data={"error": "Missing source id"}, status=404)
+
+    @extend_schema(
+        operation_id="Add a Symbol Source to a Project",
+        parameters=[GlobalParams.ORG_SLUG, GlobalParams.PROJECT_SLUG],
+        request=SourceSerializer,
+        responses={
+            201: REDACTED_SOURCE_SCHEMA,
+            400: RESPONSE_BAD_REQUEST,
+            403: RESPONSE_FORBIDDEN,
+        },
+        examples=ProjectExamples.ADD_SYMBOL_SOURCE,
+    )
+    def post(self, request: Request, project: Project) -> Response:
+        """
+        Add a custom symbol source to a project.
+        """
+        custom_symbol_sources_json = project.get_option("sentry:symbol_sources") or []
+        sources = parse_sources(custom_symbol_sources_json, False)
+
+        source = request.data
+
+        if "id" in source:
+            id = source["id"]
+        else:
+            id = str(uuid4())
+            source["id"] = id
+
+        sources.append(source)
+
+        try:
+            validate_sources(sources)
+        except InvalidSourcesError:
+            return Response(status=400)
+
+        serialized = json.dumps(sources)
+        project.update_option("sentry:symbol_sources", serialized)
+
+        redacted = redact_source_secrets([source])
+        return Response(data=redacted[0], status=201)
+
+    @extend_schema(
+        operation_id="Update a Symbol Source in a Project",
```

Comment:
I think this sounds better?
```suggestion
        operation_id="Update a Project's Symbol Source",
```